sda-commons-server-jackson/src/main/java/org/sda/commons/server/jackson/hal/HalLinkProvider.java

```diff
@@ -0,0 +1,129 @@
+package org.sda.commons.server.jackson.hal;
+
+import io.openapitools.jackson.dataformat.hal.HALLink;
+import io.openapitools.jackson.dataformat.hal.HALLink.Builder;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Feature;
+import javax.ws.rs.core.FeatureContext;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+import org.glassfish.jersey.uri.internal.JerseyUriBuilder;
+import org.sda.commons.server.jackson.hal.HalLinkInvocationStateUtility.MethodInvocationState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper utility that can be registered in the Jersey environment to be injected into services. It
```

Comment:
I wonder if we can register a singleton instance of this class at jersey in the `JacksonConfigurationBundle` and provide a static alternative of `linkTo` that delegates to the instance. What do you think?

Reply:
Could be nice if the developer doesn't need to be aware of registering in the jersey environment. 👍